alicloud/resource_alicloud_ecs_autoprovisioning_group.go

```diff
@@ -0,0 +1,394 @@
+		if v := d.Get("default_target_capacity_type"); v != nil {
+			request.DefaultTargetCapacityType = d.Get("default_target_capacity_type").(string)
+		}
+	}
+
+	if d.HasChange("terminate_instances_with_expiration") {
+		request.TerminateInstancesWithExpiration = requests.NewBoolean(d.Get("terminate_instances_with_expiration").(bool))
+	}
+
+	if d.HasChange("max_spot_price") {
+		request.MaxSpotPrice = requests.NewFloat(d.Get("max_spot_price").(float64))
+	}
+
+	if d.HasChange("total_target_capacity") {
+		if v := d.Get("total_target_capacity"); v != nil {
+			request.TotalTargetCapacity = d.Get("total_target_capacity").(string)
+		}
+	}
+
+	if d.HasChange("pay_as_you_go_target_capacity") {
+		if v := d.Get("pay_as_you_go_target_capacity"); v != nil {
+			request.PayAsYouGoTargetCapacity = d.Get("pay_as_you_go_target_capacity").(string)
+		}
+	}
+
+	if d.HasChange("spot_target_capacity") {
+		if v := d.Get("spot_target_capacity"); v != nil {
+			request.SpotTargetCapacity = d.Get("spot_target_capacity").(string)
+		}
+	}
+
+	if d.HasChange("auto_provisioning_group_name") {
+		if v := d.Get("auto_provisioning_group_name"); v != nil {
+			request.AutoProvisioningGroupName = d.Get("auto_provisioning_group_name").(string)
+		}
+	}
+
+	raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.ModifyAutoProvisioningGroup(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	d.SetPartial("excess_capacity_termination_policy")
+	d.SetPartial("default_target_capacity_type")
+	d.SetPartial("terminate_instances_with_expiration")
+	d.SetPartial("max_spot_price")
+	d.SetPartial("total_target_capacity")
+	d.SetPartial("pay_as_you_go_target_capacity")
+	d.SetPartial("spot_target_capacity")
+	d.SetPartial("auto_provisioning_group_name")
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+	d.Partial(false)
+	return resourceAlicloudEcsAutoProvisioningGroupRead(d, meta)
+}
+
+func resourceAlicloudEcsAutoProvisioningGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+
+	request := ecs.CreateDeleteAutoProvisioningGroupRequest()
+	request.RegionId = client.RegionId
+	request.AutoProvisioningGroupId = d.Id()
+	request.TerminateInstances = requests.NewBoolean(d.Get("terminate_instances").(bool))
+
+	raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DeleteAutoProvisioningGroup(request)
+	})
+
+	if err != nil {
+		if IsExpectedErrors(err, []string{"InvalidAutoProvisioningGroupId.NotFound"}) {
+			return nil
+		}
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+	return WrapError(ecsService.WaitForEcsAutoProvisioningGroup(d.Id(), DefaultTimeout))
```

Comment:
same method and parameters with create func?

Reply:
create and then describe groups, OK?